Src/tim.c

```diff
@@ -53,9 +53,9 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Init.Prescaler = 15; //(timer Frequency/(PWM STEPS * PWM FREQUENCY))-1
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 0;
+  htim16.Init.Period = 100; //PWM Steps
```

Comment:
100 does not correspond to a 50 Hz period with the prescaler you've chosen. This "period" value gets directly loaded into the auto-reload register (17.6.10 in the reference manual).

The description for that register is a bit shitty though, so also take a look at (16.3.2), which is a description of how "count up" mode, which is the mode the timer runs in for PWM, works. It should clear up how to set the period.